src/librustc_mir/const_eval/mod.rs

```diff
@@ -43,6 +45,12 @@ pub(crate) fn destructure_const<'tcx>(
 
     let variant = ecx.read_discriminant(op).unwrap().1;
```

Comment:
you can move this into the `ty::Adt` arm below, then it won't invoke `discriminant_for_variant`

Reply:
The `operand_downcast` further down can also be moved up into the `ty::Adt` arm.

I guess you'll have to either make the `variant` field become an `Option`, or just yield `VariantIdx::new(0)` from the other arms.

Reply:
Fixed!